api/envoy/extensions/filters/http/lua/v3/lua.proto

```diff
@@ -24,4 +26,40 @@ message Lua {
   // be properly escaped. YAML configuration may be easier to read since YAML supports multi-line
   // strings so complex scripts can be easily expressed inline in the configuration.
   string inline_code = 1 [(validate.rules).string = {min_bytes: 1}];
+
+  // Map of named Lua source codes that can be referenced in :ref:` LuaPerRoute
+  // <envoy_v3_api_msg_extensions.filters.http.lua.v3.LuaPerRoute>`. The Lua source codes can be
+  // loaded from inline string or local files.
+  //
+  // Example:
+  //
+  // .. code-block:: yaml
+  //
+  //   source_codes:
+  //     hello.lua:
+  //       inline_string: |
+  //         function envoy_on_response(response_handle)
+  //           -- Do something.
+  //         end
+  //     world.lua:
+  //       filename: /etc/lua/world.lua
+  //
+  map<string, config.core.v3.DataSource> source_codes = 2;
+}
+
+message LuaPerRoute {
+  oneof override {
+    option (validate.required) = true;
+
+    // Disable the Lua filter for this particular vhost or route. If disabled is specified in
+    // multiple per-filter-configs, the most specific one will be used.
+    bool disabled = 1 [(validate.rules).bool = {const: true}];
+
+    // A name of a Lua source code stored in
+    // :ref:`Lua.source_codes <envoy_v3_api_field_extensions.filters.http.lua.v3.Lua.source_codes>`.
+    string name = 2 [(validate.rules).string = {min_bytes: 1}];
+
+    // A configured per-route Lua source code that can be served by RDS or provided inline.
+    config.core.v3.DataSource source_code = 3;
```

Comment:
Shouldn't this be a oneof?

Reply:
In fact, it is  a `oneof` indeed.

Reply:
https://github.com/envoyproxy/envoy/pull/11235#discussion_r438504417

Reply:
Oops sorry nevermind, missed it above.